### DISCUSSION TITLE
Not focusing checkboxes, radios, and selects on iOS

### DISCUSSION THREAD

**Discussion by Runefarer** [react-hook-form/react-hook-form#5455] [General]
When form fields have an error, on `handleSubmit` the first error field should get focused and scrolled into view. On iOS devices, neither the scrolling nor the focusing seems to work for fields that are checkboxes, radios, or selects.

We are currently using v6 (considering migrating but it will take a while), and we depend on scrolling to the field with error on submit. But, we noticed that on iOS certain types of fields (checkboxes, radios, selects) are not focused or scrolled to.

Is this something that has been noticed before? Is there a plan to address it? If it'll be implemented in v7, that's one more reason to migrate.

From my research, it seems on iOS the scroll needs to be triggered separately. My only option, if the library doesn't, is to `useImperativeHandle` on all refs, and combine scrolling with focus.

**Comment by bluebill1049** [comment]
I believe the focus is actually there you can try with press the "space" key or tab around. I had similar issue before, because focus ring didn't appear.

**Reply by Runefarer** [reply]
Hey! Thanks for your response!

On iOS, without the on screen keyboard appearing, the user won't know which field to interact with/fix without scrolling to it.

**Reply by bluebill1049** [reply]
the focus event is not working?

**Reply by Runefarer** [reply]
I listened to `onFocus`, it did not seem to fire. This is only on iOS, for other OSs/browsers, focus and scrolling is working as intended.

**Reply by bluebill1049** [reply]
from memory, I think this is an existing WebKit bug...
